application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/NewExtensionVersionAvailableManager.java

```diff
@@ -77,26 +79,42 @@ public void checkLicensedExtensionsAvailableVersions()
             InstalledExtension installedExtension = installedRepository.getInstalledExtension(extensionId);
             Collection<String> namespaces = installedExtension.getNamespaces();
             if (namespaces == null) {
-                notifyExtensionAvailableVersions(installedExtension.getId(), null);
+                notifyExtensionAvailableVersion(installedExtension.getId(), null);
             } else {
                 for (String namespace : installedExtension.getNamespaces()) {
-                    notifyExtensionAvailableVersions(installedExtension.getId(), namespace);
+                    notifyExtensionAvailableVersion(installedExtension.getId(), namespace);
                 }
             }
         }
 
     }
 
-    private void notifyExtensionAvailableVersions(ExtensionId extensionId, String namespace)
+    private void notifyExtensionAvailableVersion(ExtensionId extensionId, String namespace)
     {
         InstalledExtension installedExtension =
             installedRepository.getInstalledExtension(extensionId.getId(), namespace);
-        List<Version> versions = upgradeExtensionHandler.getInstallableVersions(installedExtension.getId());
-        if (versions.size() > 0) {
-            String message = String.format("%s - %s - %s", installedExtension.getName(),
-                namespace != null ? namespace : "root", versions.get(0));
-            this.observationManager.notify(new NewExtensionVersionAvailableEvent(), extensionId.getId(), message);
+        List<Version> installableVersions = upgradeExtensionHandler.getInstallableVersions(installedExtension.getId());
+        if (installableVersions.size() <= 0) {
+            return;
         }
-    }
 
+        List<String> newVersionNotifiedExtensions = new ArrayList<>(licensingConfig.getNewVersionNotifiedExtensions());
+        String namespaceName = namespace != null ? namespace : "root";
+        // Create an identified for the extension, by consider also the version and namespace.
+        String verifiedExtensionId =
+            String.format("%s-%s-%s", extensionId.getId(), namespaceName, installableVersions.get(0));
```

Comment:
Why "verified"? And it would be good to mention that the first item in ``installableVersions`` is the most recent (right?).